lib/game_client/ogs/ogs_game.dart

```diff
@@ -329,6 +333,63 @@ class OGSGame extends Game {
     }
   }
 
+  void _handleClock(Map<String, dynamic> data) {
+    _logger.fine('Received clock update for game $id');
+
+    try {
+      final currentPlayer = data['current_player'] as int?;
+      final blackPlayerId = data['black_player_id'] as int?;
+      final whitePlayerId = data['white_player_id'] as int?;
+      final blackTimeData = data['black_time'] as Map<String, dynamic>?;
+      final whiteTimeData = data['white_time'] as Map<String, dynamic>?;
+
+      // Increment tick ID to signal time update
+      final currentBlackTime = blackTime.value;
+      final currentWhiteTime = whiteTime.value;
+
+      if (blackTimeData != null) {
+        final timeState = _parseOGSTimeData(blackTimeData);
+        // Use current tick ID + 1 if this is the current player, otherwise keep same tick ID
+        final isCurrentPlayer = currentPlayer == blackPlayerId;
+        final tickId =
+            isCurrentPlayer ? currentBlackTime.$1 + 1 : currentBlackTime.$1;
+        blackTime.value = (tickId, timeState);
+      }
+
+      if (whiteTimeData != null) {
+        final timeState = _parseOGSTimeData(whiteTimeData);
+        // Use current tick ID + 1 if this is the current player, otherwise keep same tick ID
+        final isCurrentPlayer = currentPlayer == whitePlayerId;
+        final tickId =
+            isCurrentPlayer ? currentWhiteTime.$1 + 1 : currentWhiteTime.$1;
+        whiteTime.value = (tickId, timeState);
+      }
+
+      _logger.fine('Updated clock for game $id: current_player=$currentPlayer');
+    } catch (error) {
+      _logger.warning('Error handling clock update for game $id: $error');
+    }
+  }
+
+  /// Helper function to safely convert dynamic value to double
+  static Duration _parseSeconds(dynamic value) {
```

Comment:
the pain of untyped stuff

Reply:
😭 even a lot of typed languages don't fail too hard on `int`->`double` cast.  ah well